File: python-playwright/src/pages/page_file_upload.py
# Created on : Apr 3, 2026, 1:24:24 PM
# Author     : Tahreem J. Naseem (www.tahreems.com / +92-333-443-8775) {TECHBYTES}

from src.core.base_page import BasePage
from src.utilities.logger import get_logger
from src.utilities.path_loader import load_uploads_dir


class PageFileUpload(BasePage):
    logger = get_logger(__name__)

    PAGE_PATH = "/upload"
    BTN_BROWSE = "#file-upload"
    BTN_UPLOAD = "#file-submit"
    SUCCESS_MESSAGE_TITLE = "#content .example h3"
    SUCCESS_MESSAGE_TEXT = "#uploaded-files"

    def open(self, base_url: str):
        self.logger.info(f"Opening File Upload page at {base_url}{self.PAGE_PATH}")
        self.navigate(f"{base_url}{self.PAGE_PATH}")

    def upload(self):
        self.logger.info("Uploading file.")
        self.upload_file(self.BTN_BROWSE, load_uploads_dir() / "uploadTestFile.txt")
        self.click_element(self.BTN_UPLOAD)

    def get_uploaded_message_title(self) -> str:
        self.logger.info("Getting uploaded message Title.")
        text = self.get_element_text(self.SUCCESS_MESSAGE_TITLE)
        if text is None:
            self.logger.error("'None' data type recieved from message Title.")
            raise ValueError("'None' data type recieved from message Title.")
        return text.strip()

    def get_uploaded_message_text(self) -> str:
        self.logger.info("Getting uploaded message Text.")
        text = self.get_element_text(self.SUCCESS_MESSAGE_TEXT)
        if text is None:
            self.logger.error("'None' data type recieved from message Text.")
            raise ValueError("'None' data type recieved from message Text.")
        return text.strip()

File: javascript-playwright/src/pages/page-file-upload.js
import path from "node:path";
import { BasePage } from "../core/base-page.js";
import { loadUploadsDir } from "../utilities/path-loader.js";

export class PageFileUpload extends BasePage {
  PAGE_PATH = "/upload";
  BTN_BROWSE = "#file-upload";
  BTN_UPLOAD = "#file-submit";
  SUCCESS_MESSAGE_TITLE = "#content .example h3";
  SUCCESS_MESSAGE_TEXT = "#uploaded-files";

  async open(baseUrl) { await this.navigate(`${baseUrl}${this.PAGE_PATH}`); }
  async uploadTestFile(fileName = "uploadTestFile.txt") {
    await this.uploadFile(this.BTN_BROWSE, path.join(loadUploadsDir(), fileName));
    await this.clickElement(this.BTN_UPLOAD);
  }
  async getUploadedMessageTitle() { return (await this.getElementText(this.SUCCESS_MESSAGE_TITLE)).trim(); }
  async getUploadedMessageText() { return (await this.getElementText(this.SUCCESS_MESSAGE_TEXT)).trim(); }
}
